anomalib/utils/callbacks/visualizer/visualizer_base.py

```diff
@@ -37,8 +38,10 @@ def __init__(
         if mode not in ["full", "simple"]:
             raise ValueError(f"Unknown visualization mode: {mode}. Please choose one of ['full', 'simple']")
         self.mode = mode
-        if task not in ["classification", "segmentation"]:
-            raise ValueError(f"Unknown task type: {mode}. Please choose one of ['classification', 'segmentation']")
+        if task not in [TaskType.CLASSIFICATION, TaskType.DETECTION, TaskType.SEGMENTATION]:
+            raise ValueError(
+                f"Unknown task type: {mode}. Please choose one of ['classification', 'detection', 'segmentation']"
```

Comment:
Not really necessary here but I wonder if `f"Unknown task type: {mode}. Please choose one of {TaskType.__members__.values()}` will work here. But will not make it a very descriptive docstring for the programmers.